app/src/main/java/doit/study/droid/InterActivityDataHolder.java

```diff
@@ -0,0 +1,19 @@
+package doit.study.droid;
+
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+
+public class InterActivityDataHolder {
+    Map<String, WeakReference<Object>> data = new HashMap<>();
```

Comment:
What's the use of this holder?
Why not just call quizData.getSomething() in "intented" class?